### ISSUE TITLE
Uh-Oh, something went wrong!

### ISSUE THREAD

**Issue by otofoto** [woj-ciech/Kamerka-GUI#14]
ERROR/MainProcess] Task handler raised error: ValueError('not enough values to unpack (expected 3, got 0)',)
Traceback (most recent call last):
  File "c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\billiard\pool.py", line 362, in workloop
    result = (True, prepare_result(fun(*args, **kwargs)))
  File "c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\celery\app\trace.py", line 546, in _fast_trace_task
    tasks, accept, hostname = _loc
ValueError: not enough values to unpack (expected 3, got 0)

**Comment by woj-ciech** [comment]
Come on, what is this? Create a proper issue with description what you did, what should be expected and what went well.

**Comment by otofoto** [comment]
Installed according to requirements and added Google Map and Shodan API key. Search can be run but when dashboard opens shows status "Starting..." and then it is replaced by "Uh-Oh, something went wrong! " error status. Looking at Shodan API key is being used but no results.

**Comment by woj-ciech** [comment]
I can't reproduce, what celery logs say?

**Comment by samsheff** [comment]
I get the same thing. Everything else works, connects to redis correctly, workers start up and site loads and submits the searches to the queue.

Once they're in the queue, it looks like it doesn't get any response from shodan and errors out. I dont think it's using the API key  since it's not taking any credits from my account (which I've added to `keys.json`, as well as the others). I've started it and set it up using the instructions in the Readme.

Have you tested it on windows? It's probably the culprit, since I'm running it there. I'd be happy to submit a PR if I can find the issue and fix it. Any hints on where to look?

Stack Trace and Worker Log:
```
celery worker -A kamerka --loglevel=info

 -------------- celery@DESKTOP-HES9I13 v4.4.5 (cliffs)
--- ***** -----
-- ******* ---- Windows-10-10.0.17763-SP0 2020-06-23 14:22:09
- *** --- * ---
- ** ---------- [config]
- ** ---------- .> app:         kamerka:0x3819950
- ** ---------- .> transport:   redis://localhost:6379//
- ** ---------- .> results:     redis://localhost:6379/
- *** --- * --- .> concurrency: 12 (prefork)
-- ******* ---- .> task events: OFF (enable -E to monitor tasks in this worker)
--- ***** -----
 -------------- [queues]
                .> celery           exchange=celery(direct) key=celery


[tasks]
  . kamerka.celery.debug_task
  . kamerka.tasks.binary_edge_scan
  . kamerka.tasks.bosch_usernames
  . kamerka.tasks.devices_nearby
  . kamerka.tasks.flickr
  . kamerka.tasks.nmap_scan
  . kamerka.tasks.send_to_field_agent_task
  . kamerka.tasks.shodan_scan_task
  . kamerka.tasks.shodan_search
  . kamerka.tasks.twitter_nearby_task
  . kamerka.tasks.whoisxml

[2020-06-23 14:22:09,915: INFO/MainProcess] Connected to redis://localhost:6379//
[2020-06-23 14:22:09,954: INFO/MainProcess] mingle: searching for neighbors
[2020-06-23 14:22:10,299: INFO/SpawnPoolWorker-1] child process 10288 calling self.run()
[2020-06-23 14:22:10,307: INFO/SpawnPoolWorker-6] child process 9152 calling self.run()
[2020-06-23 14:22:10,310: INFO/SpawnPoolWorker-4] child process 10400 calling self.run()
[2020-06-23 14:22:10,314: INFO/SpawnPoolWorker-5] child process 11424 calling self.run()
[2020-06-23 14:22:10,316: INFO/SpawnPoolWorker-7] child process 5056 calling self.run()
[2020-06-23 14:22:10,314: INFO/SpawnPoolWorker-3] child process 3292 calling self.run()
[2020-06-23 14:22:10,320: INFO/SpawnPoolWorker-2] child process 13672 calling self.run()
[2020-06-23 14:22:10,327: INFO/SpawnPoolWorker-8] child process 13940 calling self.run()
[2020-06-23 14:22:10,340: INFO/SpawnPoolWorker-10] child process 10332 calling self.run()
[2020-06-23 14:22:10,347: INFO/SpawnPoolWorker-11] child process 11576 calling self.run()
[2020-06-23 14:22:10,357: INFO/SpawnPoolWorker-9] child process 14128 calling self.run()
[2020-06-23 14:22:10,369: INFO/SpawnPoolWorker-12] child process 1624 calling self.run()
[2020-06-23 14:22:10,994: INFO/MainProcess] mingle: all alone
[2020-06-23 14:22:11,010: WARNING/MainProcess] c:\users\sam\appdata\local\programs\python\python37-32\lib\site-packages\celery\fixups\django.py:206: UserWarning: Using settings.DEBUG leads to a memory
            leak, never use this setting in production environments!
  leak, never use this setting in production environments!''')
[2020-06-23 14:22:11,011: INFO/MainProcess] celery@DESKTOP-HES9I13 ready.
[2020-06-23 14:22:21,862: INFO/MainProcess] Received task: kamerka.tasks.shodan_search[d35fbceb-7624-4127-9bf6-c0dff4a9a3c5]
[2020-06-23 14:22:21,866: ERROR/MainProcess] Task handler raised error: ValueError('not enough values to unpack (expected 3, got 0)')
Traceback (most recent call last):
  File "c:\users\sam\appdata\local\programs\python\python37-32\lib\site-packages\billiard\pool.py", line 362, in workloop
    result = (True, prepare_result(fun(*args, **kwargs)))
  File "c:\users\sam\appdata\local\programs\python\python37-32\lib\site-packages\celery\app\trace.py", line 600, in _fast_trace_task
    tasks, accept, hostname = _loc
ValueError: not enough values to unpack (expected 3, got 0)
```

**Comment by samsheff** [comment]
Update: It doesn seem to be windows specific. I setup the app in Docker/Docker Compose and it's gone in linux conatiners. Would you like me to try and make a dockerfile?

**Comment by ceyricc** [comment]
I'm having the same problem (using windows)

**Comment by nayeemx11** [comment]
@otofoto brother, you have posted for RabbitMQ
I have faced similar problem, when I am try to work with celery in Django.

The solution I have found:
https://stackoverflow.com/questions/59927934/valueerror-not-enough-values-to-unpack-expected-3-got-0-when-starting-celery

- Windows doesn't support the celery parallel processing.
- Adding --pool=solo is workaround to run the celery worker in windows.
`celery -A celery_worker.celery worker --pool=solo -l info`

for in case of Django: `celery -A [your_project_name] worker --pool=solo -l info`
